tests/components/mqtt/test_init.py

```diff
@@ -1063,6 +1063,58 @@ async def test_setup_logs_error_if_no_connect_broker(hass, caplog):
         assert "Failed to connect to MQTT server:" in caplog.text
 
 
+@patch("homeassistant.components.mqtt.TIMEOUT_ACK", 0.2)
+async def test_handle_mqtt_timeout_on_callback(hass, caplog):
+    """Test publish without receiving an ACK callback."""
+    mid = 0
+
+    class FakeInfo:
+        """Returns a simulated client publish response."""
+
+        mid = 100
+        rc = 0
+
+    with patch("paho.mqtt.client.Client") as mock_client:
+
+        @ha.callback
+        def _async_fire_mqtt_message(topic, payload, qos, retain):
+            async_fire_mqtt_message(hass, topic, payload, qos, retain)
```

Comment:
We don't need this.

Reply:
Strictly the sending of the payload can be left a way for the test. Patched the return_value now. We need this test overall to make sure all code is covered with tests.